internal/dataplane/kong_client.go

```diff
@@ -155,6 +154,9 @@ type KongClient struct {
 	// lastProcessedSnapshotHash stores the hash of the last processed Kubernetes objects cache snapshot. It's used to determine configuration
 	// changes. Please note it is always empty when the `FallbackConfiguration` feature gate is turned off.
 	lastProcessedSnapshotHash store.SnapshotHash
+
+	// brokenObjects is a list of the Kubernetes resources that failed to sync and triggered a fallback sync.
+	brokenObjects []util.ObjectHash
```

Comment:
Shouldn't it be possible to avoid caching them here? We should be able to send them to the diagnostics server as soon as we figure them out (based on the Gateway response). If it's problematic in terms of passing the broken objects around between functions, I'd suggest considering having a separate channel for fallback configuration diagnostics reports and sending them as soon as we have them available (e.g. [here](https://github.com/Kong/kubernetes-ingress-controller/blob/a2670f23f366b8e90193c785517c9399d577fda0/internal/dataplane/kong_client.go#L603)).

Reply:
As-is the diagnostic pushes are part of `sendToClient()`, and the fallback operates by maybe calling `sendToClient()` a second time, with a set of (hopefully) not-broken objects. My thinking is that we wanted to attach the broken object set to the second, successful send, though that's not really reflected on the external API endpoints. The original was designed mostly as a basic way to get generated config without tcpdump and not much thought was put into workflows using it since.  https://github.com/Kong/kubernetes-ingress-controller/issues/5932 maybe needed more info about what the fallback-related workflows would be :|